app/hookSW.js

```diff
@@ -15,24 +15,61 @@ self.addEventListener('push', function(evt) {
   console.log('Push received ', JSON.stringify(evt.data));
   var obj = evt.data ? evt.data.json() : {};
 
-  if (obj && obj.body) {
-    showNotification(obj, evt);
+  if (obj && obj.message) {
+    notifyClient(obj)
+      .then(() => {
+        showNotification(obj, evt);
+      });
   } else {
-    console.log('Discarding notification');
+    console.log('Discarding notification: ' + JSON.stringify(obj));
   }
 });
 
 self.addEventListener('pushsubscriptionchange', function(evt) {
   console.log('Got subscription change event ', evt);
 });
 
+self.addEventListener('notificationclick', function(evt) {
+  evt.notification.close();
+  evt.waitUntil(clients.matchAll({
+    type: "window"
+  }).then(function(clients) {
+    clients.forEach((client) => {
+      if ('focus' in client) {
+        client.focus();
+      }
+    });
+  }));
+});
+
+/**
+ * Notify the foxlink library of the message received via push.
+ * This will notify all the windows/tabs opened with the app.
+ * @param {object} obj Payload coming from the push notification
```

Comment:
nit1: We usually add a line break between the description and the JSDoc annotations.
nit2: We use the [JSDoc annotations as implementend in Closure Compile](https://developers.google.com/closure/compiler/docs/js-for-compiler#types), so it's `Object` (capitalised).